### DISCUSSION TITLE
Circle is 10 years old!

### DISCUSSION THREAD

**Discussion by rsta2** [rsta2/circle#491] [General]
Today Circle is exactly 10 years old. On October 25, 2014 the very [first step](https://github.com/rsta2/circle/releases/tag/Step1) of Circle was pushed to GitHub and was [announced](https://forums.raspberrypi.com/viewtopic.php?t=90130) in the Raspberry Pi bare metal forum one day later. It began with an USB driver for the Raspberry Pi Model B and B+, which needed some infrastructure. So there was the idea to release it together as a bare metal environment. In the meantime we have much more features. A number of impressive applications, appliances and support libraries have been developed by you using Circle in these years.

**Thank you to all contributors and users of Circle over this time!**

Cheers!

![Cheers!](https://github.com/user-attachments/assets/7c011e10-3825-41fe-910f-a2daeaa467ad)

**Comment by smuehlst** [comment]
Hi Rene,

congratulations on ten years with Circle! You really built a noteworthy open source project. I think it is incredible how competent and helpful you are and how you are always open for suggestions and questions.

I learnt a lot from Circle, and I'm looking forward to our collaboration in the future. 

Best regards
Stephan

**Reply by rsta2** [reply]
Hi Stephan,

thank you for your congratulations and for appreciating Circle!

Thank you also for developing and maintaining your projects circle-stdlib and circle-newlib over the time. Your projects are very important for Circle, because I think most of the Circle users are also using your libraries.

Collaboration with you has been always smooth and fun and I hope, we can continue this in the coming time.

Best regards,

Rene

**Comment by hpingel** [comment]
@rsta2 : Congratulations to sticking with this excellent project for such a long time! And also all the other side-projects and libs!

Circle is also a project that always adopted the changes that came in new with the latest hardware changes in the Raspberry Pi and had to cope with the change that the Raspberry Pi is moving further away from being a multi core MC with RAM and GPIOs to being something else. A new Raspberry has to be more than the last generation. The experience shows that for many purposes the generation 3 is still good enough to get a job done reliable with Circle.

**Reply by rsta2** [reply]
@hpingel Thank you for your congratulations and for appreciating Circle and related projects!

Yes, the Raspberry Pi platform was developed very much further in these years and I had to learn a lot to be able to support this in Circle. I also learned from you about docker/podman some time ago. This was new and very interesting for me. Thanks for that and for supporting the awesome Sidekick64 project for the C64!

**Comment by nickverlinden** [comment]
Hi Rene,

Also want to thank you for your continued efforts for such an amazing and complex project. When i first thought about jumping in on using the pi as a beefed up mcu, i started by trying to build something following David Welch's stuff and discovered Circle along the way because i was struggling with some technical stuff. Since then you helped make the dreams of an audio/visual tinkerer come true. I really think i couldn't have made most of the projects i built without Circle, or i would've given up a long time ago. Thanks so much!

Greetz!
Nick

![happy_circle](https://github.com/user-attachments/assets/12a7f5d9-1c34-4c7c-987b-fc57c83dc4a4)

**Reply by rsta2** [reply]
Hi Nick,

thank you for preparing this lovely birthday note! I'm happy that Circle was able to help you to realize your projects. I also remember, that you helped me very much to get Circle working with a number of USB Pro audio interfaces. Thank you and happy coding and tinkering!

Best regards,

Rene

**Comment by dirkarnez** [comment]
@rsta2 Omg happy birthday to your work, the amazing Circle! You never knew how this library encouraged me to learn and work with single board computers by a butterfly effect! I cannot thank you enouuuuugh!

**Reply by rsta2** [reply]
Thank you, Dirk! I'm glad, that Circle has friends in Hong Kong.

Rene

**Reply by dirkarnez** [reply]
danke schön 😊

**Comment by zsdz** [comment]
Thank you very much!
